### PR TITLE
Management of hooks folders and naming

### LINKED ISSUE #3593: Define Conan hooks management
Following the feature of Conan Plugins #3555 we should define the following points:

- Installation of plugins: `conan config install` with custom plugins parameter or normal?
- Merge strategy for plugins with same name or same subfolders.
- Automatic activation/desactivation of plugins in *conan.conf*?
- What should be the strategy to for plugin versioning?
- Should we think about a way to detect compatibility of plugins in respect to the Conan client version?

### PR DESCRIPTION
**Pull request by danimtb**
**MERGE THIS PR AFTER #4083**

Changelog: Feature: Allow hooks to be stored in folders

- [x] Refer to the issue that supports this Pull Request: closes #3593 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one: https://github.com/conan-io/docs/pull/979

### CODE REVIEW

**Comment by lasote** [comment]
This is very broken, take a look

**Comment by lasote** [comment]
Conflicted and broken, fix asap, otherwise it will be out of the release
```

======================================================================

FAIL: test_git_checkout_is_possible (blocked_v2.conans.test.command.config_install_test.ConfigInstallTest)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/tmp/source/b14a/py27/blocked_v2/conans/test/command/config_install_test.py", line 391, in test_git_checkout_is_possible

    self.assertTrue(os.path.exists(file_path))

AssertionError: False is not true

```

File: conans/client/conf/__init__.py [hunk 1]
```diff
@@ -141,7 +141,7 @@
 # no_proxy_match = *bintray.com*, https://myserver.*
 
 [hooks]    # environment CONAN_HOOKS
-attribute_checker
+attribute_checker/attribute_checker.py
```
Comment (memsharded) [inline conans/client/conf/__init__.py]: This will fail in an upgrade?

Comment (memsharded) [reply conans/client/conf/__init__.py]: I think I will leave it as-is, it is ok to have hooks also not in folders, isn't it?

File: conans/client/conf/config_installer.py [hunk 2]
```diff
@@ -182,3 +188,21 @@ def _process_config_install_item(item):
         path_or_url, args = [item.strip() for item in rest[1:-1].rstrip().split(",", 1)]
         args = None if "none" in args.lower() else args
     return config_type, path_or_url, args
+
+
+def _handle_hooks(src_hooks_path, dst_hooks_path, output):
```
Comment (lasote) [inline conans/client/conf/config_installer.py]: This is only to skip the .git folders or what?

File: conans/client/conf/__init__.py (hunk 1 above)
Comment (lasote) [reply conans/client/conf/__init__.py]: It doesn't fail, I've tried it, but eventually, you end with the `attribute_checker.py` in the root and also i the `attribute_checker/attribute_checker.py`. Maybe the Conan ones are ok in the root to simplify

File: conans/client/conf/config_installer.py (hunk 2 above)
Comment (danimtb) [reply conans/client/conf/config_installer.py]: Yes, there is a *hooks* folder in the .git folder, so I prefer to have this function to isolate the copy of the files

**Pull request merged**